src/backend/metal/Cargo.toml

```diff
@@ -34,7 +34,7 @@ foreign-types = "0.3"
 objc = "0.2.5"
 block = "0.1"
 cocoa-foundation = "0.1"
-spirv_cross = { version = "0.22", features = ["msl"] }
+spirv_cross = { version = "0.22", features = ["msl"], git = "https://github.com/grovesNL/spirv_cross", rev = "61d3b8c" }
```

Comment:
this seems unnecessary

Reply:
without these changes I cannot compile with the latest version of spirv_cross.
`Ast` don't has the `Parse` and `Compile` traits implemented.

Reply:
That's weird. We build this on CI for every revision, so it compiles.
I just double-checked on my macOS, running `make` in the root of latest master works fine.

Reply:
Yes it works because you don't build for web, and the master branch of gfx don't support web.
And it's fixed in only [1 commit](https://github.com/grovesNL/spirv_cross/commit/cf025136927c89070b82a1b0f4e9d3d7df806b9e#diff-dc54eb49a875a14243b80fbefe0fb34ddcd9576f674edd0f3e6d1eef3f380f2e) ...

Reply:
This is the Metal backend. We aren't supposed to build it on WASM at all.

Reply:
Yes sorry.
I need the last commit for GL backend to fix the Wasm target.
But GL Backend depends on the auxil, that requires to be updated like other backend ...

To limit changes, I can update with:
- last commit of `spirv-cross` for `backend-gl`
- last commit of `spirv-cross` for `auxil` only for `gl` feature

But is it really something relevant ... I'm not 100% convinced.

Reply:
that's why I have added the check on the issue `use published version of spirv_cross dependency`
So maybe we can wait for the new release of spirv before merging that PR.

Reply:
Hi @grovesNL, @kvark,
 I have tested locally and personally I'm not 100% sure.
We have 2 options to work with patch:
```toml
[dependencies]
# ...
spirv_cross = { version = "0.22", features = ["glsl"]}
# ...

[patch.'https://github.com/grovesNL/spirv_cross']
spirv_cross = { git = 'https://github.com/grovesNL/spirv_cross', rev = '61d3b8c' }
```

Or that I have used:
```toml
spirv_cross = { version = "0.22", features = ["glsl"], git = 'https://github.com/grovesNL/spirv_cross', rev = '61d3b8c'}
```

Personnaly I'm not a very fan of `[patch....]` syntax.
Do you think to a better solution ?

And in addition, that don't works:
```toml
[target.'cfg(target_arch = "wasm32")'.patch.'https://github.com/grovesNL/spirv_cross']
spirv_cross = { git = 'https://github.com/grovesNL/spirv_cross', rev = '61d3b8c' }
```

So every backend and auxil has to have the patch information...

Reply:
Yeah that's right, I just meant it's possible to use `patch` locally to test the fix out, instead of replacing all of the dependencies.

I [published a new version of spirv_cross](https://crates.io/crates/spirv_cross) so we can leave it as `version = "0.22"` for now (`cargo update` should automatically update to `0.22.1`)